allocating using module variable and using module routine outside module throws error

> mms_copy.f90

```fortran
MODULE XX
  IMPLICIT NONE

  INTEGER, PUBLIC :: nx = 1000

END MODULE

MODULE mms_module

  USE XX, ONLY: nx

  IMPLICIT NONE
  REAL, ALLOCATABLE, DIMENSION(:), PUBLIC :: ref_flux

  REAL, ALLOCATABLE, DIMENSION(:), PUBLIC :: ref_fluxm

  CONTAINS

  SUBROUTINE mms_deallocate
  END SUBROUTINE mms_deallocate

  SUBROUTINE mms_allocate ( )
    ALLOCATE( ref_flux(nx), ref_fluxm(nx) )
  END SUBROUTINE mms_allocate
END MODULE mms_module
```

> dealloc.f90


```fortran
MODULE dealloc_module
  USE mms_module, ONLY: mms_deallocate
END MODULE dealloc_module
```

```console
% lfortran -c mms_copy.f90 && lfortran -c dealloc.f90
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 2459
    LCompilers::print_stack_on_segfault();
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 2391
    return compile_to_object_file(arg_file, outfile, false,
  File "/Users/pranavchiku/repos/lfortran/src/bin/lfortran.cpp", line 906
    result = fe.get_asr2(input, lm, diagnostics);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/fortran_evaluator.cpp", line 252
    Result<ASR::TranslationUnit_t*> res2 = get_asr3(*ast, diagnostics);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/fortran_evaluator.cpp", line 274
    compiler_options.symtab_only, compiler_options);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_to_asr.cpp", line 77
    if (res.ok) {
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 3457
    v.visit_TranslationUnit(ast);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 120
    visit_ast(*x.m_items[i]);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4878
    void visit_ast(const ast_t &b) { visit_ast_t(b, self()); }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4834
    case astType::unit: { v.visit_unit((const unit_t &)x); return; }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4881
    void visit_mod(const mod_t &b) { visit_mod_t(b, self()); }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4465
    case modType::Module: { v.visit_Module((const Module_t &)x); return; }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 471
    visit_ModuleSubmoduleCommon<AST::Module_t, ASR::Module_t>(x);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 424
    visit_unit_decl1(*x.m_use[i]);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4890
    void visit_unit_decl1(const unit_decl1_t &b) { visit_unit_decl1_t(b, self()); }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast.h", line 4486
    case unit_decl1Type::Use: { v.visit_Use((const Use_t &)x); return; }
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 2565
    t = (ASR::symbol_t*)(ASRUtils::load_module(al, tu_symtab,
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_utils.cpp", line 308
    *symtab, intrinsic, pass_options);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_utils.cpp", line 456
    ASR::TranslationUnit_t *asr = load_modfile(al, modfile, false, symtab);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/modfile.cpp", line 93
    ASR::asr_t *asr = deserialize_asr(al, asr_binary, load_symtab_id, symtab);
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast_serialization.cpp", line 414
    Line not found
  File "/Users/pranavchiku/repos/lfortran/src/lfortran/ast_serialization.cpp", line 430
    Line not found
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 1242
    v.visit_TranslationUnit(unit);
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 111
    for (auto &a : x.m_symtab->get_scope()) {
  File "../libasr/asr.h", line 5143
  File "../libasr/asr.h", line 4855
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 333
    for (auto &a : x.m_symtab->get_scope()) {
  File "../libasr/asr.h", line 5143
  File "../libasr/asr.h", line 4856
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 454
    visit_stmt(*x.m_body[i]);
  File "../libasr/asr.h", line 5160
  File "../libasr/asr.h", line 4878
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 1218
    BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
  File "../libasr/asr.h", line 5477
  File "../libasr/asr.h", line 6619
  File "/Users/pranavchiku/repos/lfortran/src/libasr/asr_verify.cpp", line 1171
    require_with_loc(ASRUtils::is_integer(
  File "../libasr/asr_utils.h", line 294
  File "../libasr/asr.h", line 50299
AssertFailed: e->m_external
```

Another one when compiling `output.f90`

```console
(lf) saurabh-kumar@Awadh:~/Projects/Packages/SNAP/src$ make
lfortran  -c setup.f90
lfortran  -c output.f90
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran", in _start()
  File "./csu/../csu/libc-start.c", line 360, in __libc_start_main_impl()
  File "./csu/../sysdeps/x86/libc-start.c", line 58, in __libc_start_call_main()
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 2462, in main()
    return main_app(argc, argv);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 2391, in main_app(int, char**)
    return compile_to_object_file(arg_file, outfile, false,
  File "/home/saurabh-kumar/Projects/System/lfortran/src/bin/lfortran.cpp", line 906, in compile_to_object_file()
    result = fe.get_asr2(input, lm, diagnostics);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/fortran_evaluator.cpp", line 252, in LCompilers::FortranEvaluator::get_asr2(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::LocationManager&, LCompilers::diag::Diagnostics&)
    Result<ASR::TranslationUnit_t*> res2 = get_asr3(*ast, diagnostics);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/fortran_evaluator.cpp", line 274, in LCompilers::FortranEvaluator::get_asr3(LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&)
    compiler_options.symtab_only, compiler_options);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_to_asr.cpp", line 76, in LCompilers::LFortran::ast_to_asr(Allocator&, LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&, LCompilers::SymbolTable*, bool, LCompilers::CompilerOptions&)
    instantiate_types, instantiate_symbols, entry_functions, entry_function_arguments_mapping, data_structure);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 3457, in LCompilers::LFortran::symbol_table_visitor(Allocator&, LCompilers::LFortran::AST::TranslationUnit_t&, LCompilers::diag::Diagnostics&, LCompilers::SymbolTable*, LCompilers::CompilerOptions&, std::map<unsigned long, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > >, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > > > > >&, std::map<unsigned long, LCompilers::ASR::symbol_t*, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, LCompilers::ASR::symbol_t*> > >&, std::map<unsigned long, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > >&, std::map<unsigned int, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::ttype_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::ttype_t*> > > > > >&, std::map<unsigned int, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::symbol_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::symbol_t*> > >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, LCompilers::ASR::symbol_t*, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, LCompilers::ASR::symbol_t*> > > > > >&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> > > > >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<LCompilers::LFortran::AST::stmt_t*, std::allocator<LCompilers::LFortran::AST::stmt_t*> > > > > > > >&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<int, std::allocator<int> > > > >&, std::vector<LCompilers::ASR::stmt_t*, std::allocator<LCompilers::ASR::stmt_t*> >&)
    v.visit_TranslationUnit(ast);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 120, in LCompilers::LFortran::SymbolTableVisitor::visit_TranslationUnit(LCompilers::LFortran::AST::TranslationUnit_t const&)
    visit_ast(*x.m_items[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4878, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::SymbolTableVisitor>::visit_ast(LCompilers::LFortran::AST::ast_t const&)
    void visit_ast(const ast_t &b) { visit_ast_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4835, in visit_ast_t<LCompilers::LFortran::SymbolTableVisitor>()
    case astType::mod: { v.visit_mod((const mod_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4881, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::SymbolTableVisitor>::visit_mod(LCompilers::LFortran::AST::mod_t const&)
    void visit_mod(const mod_t &b) { visit_mod_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4465, in visit_mod_t<LCompilers::LFortran::SymbolTableVisitor>()
    case modType::Module: { v.visit_Module((const Module_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 471, in LCompilers::LFortran::SymbolTableVisitor::visit_Module(LCompilers::LFortran::AST::Module_t const&)
    visit_ModuleSubmoduleCommon<AST::Module_t, ASR::Module_t>(x);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 424, in void LCompilers::LFortran::SymbolTableVisitor::visit_ModuleSubmoduleCommon<LCompilers::LFortran::AST::Module_t, LCompilers::ASR::Module_t>(LCompilers::LFortran::AST::Module_t const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
    visit_unit_decl1(*x.m_use[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4890, in LCompilers::LFortran::AST::BaseVisitor<LCompilers::LFortran::SymbolTableVisitor>::visit_unit_decl1(LCompilers::LFortran::AST::unit_decl1_t const&)
    void visit_unit_decl1(const unit_decl1_t &b) { visit_unit_decl1_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/ast.h", line 4486, in visit_unit_decl1_t<LCompilers::LFortran::SymbolTableVisitor>()
    case unit_decl1Type::Use: { v.visit_Use((const Use_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp", line 2565, in LCompilers::LFortran::SymbolTableVisitor::visit_Use(LCompilers::LFortran::AST::Use_t const&)
    t = (ASR::symbol_t*)(ASRUtils::load_module(al, tu_symtab,
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_utils.cpp", line 307, in LCompilers::ASRUtils::load_module(Allocator&, LCompilers::SymbolTable*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::Location const&, bool, LCompilers::PassOptions&, bool, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::Location const&)>)
    ASR::TranslationUnit_t *mod1 = find_and_load_module(al, module_name,
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_utils.cpp", line 456, in LCompilers::ASRUtils::find_and_load_module(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, LCompilers::SymbolTable&, bool, LCompilers::PassOptions&)
    ASR::TranslationUnit_t *asr = load_modfile(al, modfile, false, symtab);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/modfile.cpp", line 93, in LCompilers::load_modfile(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, LCompilers::SymbolTable&)
    ASR::asr_t *asr = deserialize_asr(al, asr_binary, load_symtab_id, symtab);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/serialization.cpp", line 414, in LCompilers::deserialize_asr(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, LCompilers::SymbolTable&)
    return deserialize_asr(al, s, load_symtab_id);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/serialization.cpp", line 430, in LCompilers::deserialize_asr(Allocator&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)
    if (!asr_verify(*tu, false, diagnostics)) {
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 1242, in LCompilers::asr_verify(LCompilers::ASR::TranslationUnit_t const&, bool, LCompilers::diag::Diagnostics&)
    v.visit_TranslationUnit(unit);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 112, in LCompilers::ASR::VerifyVisitor::visit_TranslationUnit(LCompilers::ASR::TranslationUnit_t const&)
    this->visit_symbol(*a.second);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5143, in LCompilers::ASR::BaseVisitor<LCompilers::ASR::VerifyVisitor>::visit_symbol(LCompilers::ASR::symbol_t const&)
    void visit_symbol(const symbol_t &b) { visit_symbol_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 4856, in visit_symbol_t<LCompilers::ASR::VerifyVisitor>()
    case symbolType::Module: { v.visit_Module((const Module_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 334, in LCompilers::ASR::VerifyVisitor::visit_Module(LCompilers::ASR::Module_t const&)
    this->visit_symbol(*a.second);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5143, in LCompilers::ASR::BaseVisitor<LCompilers::ASR::VerifyVisitor>::visit_symbol(LCompilers::ASR::symbol_t const&)
    void visit_symbol(const symbol_t &b) { visit_symbol_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 4857, in visit_symbol_t<LCompilers::ASR::VerifyVisitor>()
    case symbolType::Function: { v.visit_Function((const Function_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 454, in LCompilers::ASR::VerifyVisitor::visit_Function(LCompilers::ASR::Function_t const&)
    visit_stmt(*x.m_body[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5160, in LCompilers::ASR::BaseVisitor<LCompilers::ASR::VerifyVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 4893, in visit_stmt_t<LCompilers::ASR::VerifyVisitor>()
    case stmtType::If: { v.visit_If((const If_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5558, in LCompilers::ASR::BaseWalkVisitor<LCompilers::ASR::VerifyVisitor>::visit_If(LCompilers::ASR::If_t const&)
    self().visit_stmt(*x.m_body[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5160, in LCompilers::ASR::BaseVisitor<LCompilers::ASR::VerifyVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 4893, in visit_stmt_t<LCompilers::ASR::VerifyVisitor>()
    case stmtType::If: { v.visit_If((const If_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5558, in LCompilers::ASR::BaseWalkVisitor<LCompilers::ASR::VerifyVisitor>::visit_If(LCompilers::ASR::If_t const&)
    self().visit_stmt(*x.m_body[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5160, in LCompilers::ASR::BaseVisitor<LCompilers::ASR::VerifyVisitor>::visit_stmt(LCompilers::ASR::stmt_t const&)
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 4878, in visit_stmt_t<LCompilers::ASR::VerifyVisitor>()
    case stmtType::Allocate: { v.visit_Allocate((const Allocate_t &)x); return; }
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 1218, in LCompilers::ASR::VerifyVisitor::visit_Allocate(LCompilers::ASR::Allocate_t const&)
    BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 5477, in LCompilers::ASR::BaseWalkVisitor<LCompilers::ASR::VerifyVisitor>::visit_Allocate(LCompilers::ASR::Allocate_t const&)
    self().visit_alloc_arg(x.m_args[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr.h", line 6619, in LCompilers::ASR::BaseWalkVisitor<LCompilers::ASR::VerifyVisitor>::visit_alloc_arg(LCompilers::ASR::alloc_arg_t const&)
    self().visit_dimension(x.m_dims[i]);
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/asr_verify.cpp", line 1171, in LCompilers::ASR::VerifyVisitor::visit_dimension(LCompilers::ASR::dimension_t const&)
    require_with_loc(ASRUtils::is_integer(
  File "/home/saurabh-kumar/Projects/System/lfortran/src/libasr/../libasr/asr_utils.h", line 294, in expr_type()
    return ASR::expr_type0(f);
AssertFailed: e->m_external
make: *** [Makefile:176: output.o] Error 1
```

--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -889,6 +889,7 @@ RUN(NAME modules_54 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME modules_55 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME modules_56 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc NOFAST)
 RUN(NAME modules_57 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
+RUN(NAME modules_58 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRAFILES modules_58_module.f90)
 RUN(NAME operator_overloading_05_module3 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc EXTRAFILES
         operator_overloading_05_module1.f90 operator_overloading_05_module2.f90)
 RUN(NAME associate_06 LABELS gfortran EXTRAFILES

--- a/integration_tests/modules_58.f90
+++ b/integration_tests/modules_58.f90
@@ -1,0 +1,5 @@
+program test_external_symbol
+    use module_58_module02
+    call mms_allocate()
+    call test_alocate_external
+end program test_external_symbol

--- a/integration_tests/modules_58_module.f90
+++ b/integration_tests/modules_58_module.f90
@@ -1,0 +1,38 @@
+MODULE module_58_module01
+    IMPLICIT NONE
+  
+    INTEGER, PUBLIC :: nx1 = 1000
+    INTEGER, PUBLIC :: nx2 = 1000
+    REAL, ALLOCATABLE, DIMENSION(:), PUBLIC :: arr
+  
+  END MODULE
+  
+  MODULE module_58_module02
+  
+    USE module_58_module01
+  
+    IMPLICIT NONE
+    REAL, ALLOCATABLE, DIMENSION(:), PUBLIC :: ref_flux
+  
+    REAL, ALLOCATABLE, DIMENSION(:), PUBLIC :: ref_fluxm
+  
+    CONTAINS
+  
+    SUBROUTINE mms_deallocate
+    END SUBROUTINE mms_deallocate
+  
+    SUBROUTINE mms_allocate ( )
+        nx1 = 555
+      ALLOCATE( ref_flux(nx1), ref_fluxm(nx2) )
+      print *,"Size of nx1",nx1
+      if(size(ref_flux) /= 555) error stop
+      print *,"Size of nx2",nx2
+      if(size(ref_fluxm) /= 1000) error stop
+    END SUBROUTINE mms_allocate
+
+    SUBROUTINE test_alocate_external
+      allocate(arr(nx1))
+      print *, size(arr)
+      if(size(ref_flux) /= 555) error stop
+    end SUBROUTINE test_alocate_external
+  END MODULE module_58_module02

--- a/src/libasr/asr_verify.cpp
+++ b/src/libasr/asr_verify.cpp
@@ -1222,8 +1222,8 @@ public:
                     "Allocate should only be called with  Allocatable or Pointer type inputs, found " +
                     std::string(ASRUtils::get_type_code(ASRUtils::expr_type(x.m_args[i].m_a))));
             }
-            BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
         }
+        BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
     }
 
     void visit_DoConcurrentLoop(const DoConcurrentLoop_t &x) {

--- a/src/libasr/asr_verify.cpp
+++ b/src/libasr/asr_verify.cpp
@@ -890,11 +890,13 @@ public:
             require(x.m_new != x.m_old, "ArrayPhysicalCast is redundant, "
                 "the old physical type and new physical type must be different.");
         }
-        require(x.m_new == ASRUtils::extract_physical_type(x.m_type),
-            "Destination physical type conflicts with the physical type of target");
-        require(x.m_old == ASRUtils::extract_physical_type(ASRUtils::expr_type(x.m_arg)),
-            "Old physical type conflicts with the physical type of argument " + std::to_string(x.m_old)
-            + " " + std::to_string(ASRUtils::extract_physical_type(ASRUtils::expr_type(x.m_arg))));
+        if(check_external){
+            require(x.m_new == ASRUtils::extract_physical_type(x.m_type),
+                "Destination physical type conflicts with the physical type of target");
+            require(x.m_old == ASRUtils::extract_physical_type(ASRUtils::expr_type(x.m_arg)),
+                "Old physical type conflicts with the physical type of argument " + std::to_string(x.m_old)
+                + " " + std::to_string(ASRUtils::extract_physical_type(ASRUtils::expr_type(x.m_arg))));
+        }
     }
 
     void visit_SubroutineCall(const SubroutineCall_t &x) {
@@ -1161,16 +1163,20 @@ public:
 
     void visit_dimension(const dimension_t &x) {
         if (x.m_start) {
-            require_with_loc(ASRUtils::is_integer(
-                *ASRUtils::expr_type(x.m_start)),
-                "Start dimension must be a signed integer", x.loc);
+            if(check_external){
+                require_with_loc(ASRUtils::is_integer(
+                    *ASRUtils::expr_type(x.m_start)),
+                    "Start dimension must be a signed integer", x.loc);
+            }
             visit_expr(*x.m_start);
         }
 
         if (x.m_length) {
-            require_with_loc(ASRUtils::is_integer(
-                *ASRUtils::expr_type(x.m_length)),
-                "Length dimension must be a signed integer", x.loc);
+            if(check_external){
+                require_with_loc(ASRUtils::is_integer(
+                    *ASRUtils::expr_type(x.m_length)),
+                    "Length dimension must be a signed integer", x.loc);
+            }
             visit_expr(*x.m_length);
         }
     }
@@ -1209,13 +1215,15 @@ public:
     }
 
     void visit_Allocate(const Allocate_t &x) {
-        for( size_t i = 0; i < x.n_args; i++ ) {
-            require(ASR::is_a<ASR::Allocatable_t>(*ASRUtils::expr_type(x.m_args[i].m_a)) ||
-                    ASR::is_a<ASR::Pointer_t>(*ASRUtils::expr_type(x.m_args[i].m_a)),
-                "Allocate should only be called with  Allocatable or Pointer type inputs, found " +
-                std::string(ASRUtils::get_type_code(ASRUtils::expr_type(x.m_args[i].m_a))));
+        if(check_external){
+            for( size_t i = 0; i < x.n_args; i++ ) {
+                require(ASR::is_a<ASR::Allocatable_t>(*ASRUtils::expr_type(x.m_args[i].m_a)) ||
+                        ASR::is_a<ASR::Pointer_t>(*ASRUtils::expr_type(x.m_args[i].m_a)),
+                    "Allocate should only be called with  Allocatable or Pointer type inputs, found " +
+                    std::string(ASRUtils::get_type_code(ASRUtils::expr_type(x.m_args[i].m_a))));
+            }
+            BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
         }
-        BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
     }
 
     void visit_DoConcurrentLoop(const DoConcurrentLoop_t &x) {

